specification/streamanalytics/resource-manager/Microsoft.StreamAnalytics/stable/2020-03-01/inputs.json

```diff
@@ -0,0 +1,1084 @@
+        "error": {
+          "readOnly": true,
+          "$ref": "#/definitions/ErrorResponse",
+          "description": "Describes the error that occurred."
+        }
+      }
+    },
+    "ErrorResponse": {
+      "description": "Describes the error that occurred.",
+      "properties": {
+        "code": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Error code associated with the error that occurred."
+        },
+        "message": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Describes the error in detail."
+        }
+      }
+    },
+    "SubResource": {
+      "description": "The base sub-resource model definition.",
+      "properties": {
+        "id": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource Id"
+        },
+        "name": {
+          "type": "string",
+          "description": "Resource name"
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource type"
+        }
+      },
+      "x-ms-azure-resource": true
+    },
+    "Compression": {
+      "description": "Describes how input data is compressed",
+      "properties": {
+        "type": {
+          "$ref": "#/definitions/CompressionType"
+        }
+      },
+      "required": [
+        "type"
+      ]
+    },
+    "CompressionType": {
+      "type": "string",
+      "description": "Indicates the type of compression that the input uses. Required on PUT (CreateOrReplace) requests.",
+      "enum": [
+        "None",
+        "GZip",
+        "Deflate"
+      ],
+      "x-ms-enum": {
+        "name": "CompressionType",
+        "modelAsString": true
+      }
+    },
+    "AzureSqlReferenceInputDataSource": {
+      "description": "Describes an Azure SQL database reference input data source.",
+      "x-ms-discriminator-value": "Microsoft.Sql/Server/Database",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ReferenceInputDataSource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/AzureSqlReferenceInputDataSourceProperties"
+        }
+      }
+    },
+    "AzureSqlReferenceInputDataSourceProperties": {
+      "properties": {
+        "server": {
+          "description": "This element is associated with the datasource element. This is the name of the server that contains the database that will be written to.",
+          "type": "string"
+        },
+        "database": {
+          "description": "This element is associated with the datasource element. This is the name of the database that output will be written to.",
+          "type": "string"
+        },
+        "user": {
+          "description": "This element is associated with the datasource element. This is the user name that will be used to connect to the SQL Database instance.",
+          "type": "string"
+        },
+        "password": {
```

Comment:
please mark as a secret and ensure it isn't returned on responses or GETs

Reply:
In general, we are following the rule that no secrets are returned in any response payloads. We currently do not have any APIs that will return these secrets once you set them.